server/services/discovery/enrichment/kubernetes_enrichment.py

```
@@ -422,18 +472,39 @@ def _discover_cluster(cluster, provider_credentials):
+    # authenticator uses the right credentials on every kubectl call.
+    if provider == "aws" and provider_envs.get("_aws_multi"):
+        multi_envs = provider_envs["_aws_multi"]
+        arn = cluster.get("cloud_resource_id", "")
+        parts = arn.split(":") if arn.startswith("arn:aws") else []
+        acct_id = parts[4] if len(parts) >= 5 and parts[4] else None
+        kubectl_env = multi_envs.get(acct_id, kubectl_env)
 
-    # Step 2: Fetch all resource types via kubectl
     raw_resources = {}
     for resource_kind, cmd in KUBECTL_COMMANDS.items():
         logger.info(f"K8s enrichment: fetching {resource_kind} from cluster {cluster_name}")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)